src/components/Post/Post.tsx

```diff
@@ -1,32 +1,52 @@
 import React, { FC } from 'react';
-import { getTwitterId } from '../../util';
+import { getTwitterId, getIgId } from '../../util';
 import Tweet from './Tweet';
+import InstagramEmbed from './Instagram';
 import './Post.css';
 
 type Props = {
-  onComplete?: (tweetId: string) => void;
-  onReport: (tweetId: string) => void;
+  onComplete?: (postId: string) => void;
+  onReport: (postId: string) => void;
   post: Post;
   userUid: string;
 };
 
 const Post: FC<Props> = (props: Props) => {
-  const tweetId = getTwitterId(props.post.tweetUrl);
+  const postId = getTwitterId(props.post.postUrl) || getIgId(props.post.postUrl);
 
   if (
-    !tweetId ||
+    !postId ||
     !props.post.approvedAt ||
     props.post.reporters?.includes(props.userUid)
   ) {
     return null;
   }
 
-  const handleComplete = () => props.onComplete?.(tweetId);
-  const handleReport = () => props.onReport(tweetId);
+  const handleComplete = () => props.onComplete?.(postId);
+  const handleReport = () => props.onReport(postId);
 
+  if (props.post.postUrl.startsWith('https://twitter.com')) {
+    return (
+      <div className="post" id={props.post.id}>
+        <Tweet tweetId={postId} tweetUrl={props.post.postUrl} />
+        <div className="post-buttons">
+          {props.onComplete && (
+            <button className="button is-success" onClick={handleComplete}>
+              Mark Complete
+            </button>
+          )}
+          <button className="button is-danger" onClick={handleReport}>
+            Report
+          </button>
+        </div>
+      </div>
+    );
+  }
+
+  // if (props.post.postUrl.startsWith('https://instagram.com')) {
```

Comment:
Enabling this condition causes a compilation error because it thinks `const Post` returns undefined. Not sure if this is the best way to go about choosing post types.